$NetBSD: patch-aa,v 1.6 2019/09/02 08:51:29 christos Exp $

- configure for pkgsrc
- honor PKGMANDIR
- honor pkgsrc CFLAGS and LDFLAGS
- add -DUSE_OPENPTY -lutil

--- Makefile.orig	2015-10-16 15:29:50.000000000 -0400
+++ Makefile	2019-09-02 04:47:15.636767739 -0400
@@ -21,12 +21,12 @@
 # If they don't exist, this makefile will try to create the directories
 # LIBDIR and SHAREDIR.  All others must already exist.
 
-JOVEHOME = /usr/local
-SHAREDIR = $(JOVEHOME)/lib/jove
+JOVEHOME = $(PREFIX)
+SHAREDIR = $(JOVEHOME)/share/jove
 LIBDIR = $(JOVEHOME)/lib/jove
 BINDIR = $(JOVEHOME)/bin
 XEXT=
-MANDIR = $(JOVEHOME)/man/man$(MANEXT)
+MANDIR = $(JOVEHOME)/$(PKGMANDIR)/man$(MANEXT)
 MANEXT = 1
 
 # TMPDIR is where the tmp files get stored, usually /tmp, /var/tmp, or
@@ -38,11 +38,11 @@
 # (in case the system startup salvages tempfiles by moving them,
 # which is probably a good idea).
 
-TMPDIR = /tmp
+TMPDIR = /var/tmp
 RECDIR = /var/preserve
 
 # DFLTSHELL is the default shell invoked by JOVE and TEACHJOVE.
-DFLTSHELL = /bin/csh
+DFLTSHELL = /bin/sh
 
 # The install commands of BSD and System V differ in unpleasant ways:
 # -c: copy (BSD); -c dir: destination directory (SysV)
@@ -53,15 +53,15 @@
 # "cp" will work reasonably well, but be aware that any links continue
 # referencing the old file with new contents.
 
-INSTALLFLAGS = # -g bin -o root
+INSTALLFLAGS = -g $(ROOT_GROUP) -o $(ROOT_USER)
 
 # to install executable files
-XINSTALL=cp
+XINSTALL=$(BSD_INSTALL_PROGRAM)
 #XINSTALL=/usr/ucb/install $(INSTALLFLAGS) -c -m 755 # -s
 #CYGWIN32: XINSTALL=install $(INSTALLFLAGS) -c -m 755
 
 # to install text files
-TINSTALL=cp
+TINSTALL=$(BSD_INSTALL_DATA)
 #TINSTALL=/usr/ucb/install $(INSTALLFLAGS) -c -m 644
 #CYGWIN32: TINSTALL=install $(INSTALLFLAGS) -c -m 644
 
@@ -137,7 +137,7 @@
 # Add -DUSE_EXIT if you're profiling or using purify (this causes Jove
 # to exit using exit(), instead of _exit()).
 
-SYSDEFS = -DBSDPOSIX_STDC
+SYSDEFS = -DBSDPOSIX_STDC -DUSE_OPENPTY
 
 # Select optimization level (flags passed to compiling and linking steps).
 # On most systems: -g for debugging, -O for optimization.
@@ -146,7 +146,7 @@
 # On DEC OSF/1 and Digital UNIX VV4.0, add -std1 to enable ANSI C features
 # and perhaps -g3 for more debugging info with optimization.
 
-OPTFLAGS = -O
+OPTFLAGS += -O2
 
 # For making dependencies under BSD systems
 DEPENDFLAG = -M
@@ -169,7 +169,7 @@
 #	FreeBSD 4.2: EXTRALIBS = -lutil
 #	FreeBSD 4.2, NetBSD 1.5, OpenBSD 2.x:  EXTRALIBS = -lutil
 
-EXTRALIBS =
+EXTRALIBS = -lutil
 
 # Flags of linker (LDFLAGS)
 # Most systems do not need any flags.
@@ -182,13 +182,13 @@
 #	PDP-11 with separate I&D: -i
 #	PDP-11 without separate I&D: -n
 
-LDFLAGS =
+LDFLAGS+=
 
 # for SCO Xenix, set
 #	MEMFLAGS = -Mle
 #	CFLAGS = -LARGE -O -F 3000 -K -Mle  (say -Mle2 for an 80286)
 
-CFLAGS = $(OPTFLAGS) $(SYSDEFS)
+CFLAGS+= $(OPTFLAGS) $(SYSDEFS)
 
 # For SYSVR4 (/usr/ucb/cc will NOT work because of setjmp.h):
 #	CC = /usr/bin/cc
